File: src/main/java/com/SWEProject/Entities/Store.java
package com.SWEProject.Entities;

import java.util.HashSet;
import java.util.Set;

import javax.persistence.CascadeType;
import javax.persistence.Entity;
import javax.persistence.GeneratedValue;
import javax.persistence.GenerationType;
import javax.persistence.Id;
import javax.persistence.ManyToMany;
import javax.persistence.ManyToOne;
import javax.persistence.OneToMany;

@Entity
public class Store {

	@Id
	@GeneratedValue(strategy = GenerationType.AUTO)
	private Integer id;
	private String name;
	private String type;
	private String location;
	private Boolean onSystem;
	
	@OneToMany(targetEntity= StoreProducts.class, cascade= CascadeType.ALL)
	private Set<StoreProducts> storeProducts;

	public Set<StoreProducts> getStoreProducts() {
		return storeProducts;
	}

	public void setStoreProducts(Set<StoreProducts> storeProducts) {
		this.storeProducts = storeProducts;
	}

	@ManyToOne
	private User owner;
	
	@ManyToMany(targetEntity = User.class, cascade = CascadeType.ALL)
	private Set<User> collaborators;
	
	@OneToMany(targetEntity = History.class, cascade = CascadeType.ALL)
	private Set<History> operations;
	
	public Store() {
		this.name = "";
		this.type = "";
		this.location = "";
		this.owner = null;
		this.onSystem = false;
		this.storeProducts = new HashSet<StoreProducts>();
		this.collaborators = new HashSet<User>();
		this.operations = new HashSet<History>();
	}

	public Store(String name, String type, String location, Boolean onSystem, User owner,
			Set<StoreProducts> products, Set<User> collaborators, Set<History> operations) {
		super();
		this.name = name;
		this.type = type;
		this.location = location;
		this.onSystem = onSystem;
		this.owner = owner;
		this.storeProducts = products;
		this.collaborators = collaborators;
		this.operations = operations;
	}

	public Integer getId() {
		return id;
	}

	public void setId(Integer id) {
		this.id = id;
	}

	public String getName() {
		return name;
	}

	public void setName(String name) {
		this.name = name;
	}

	public String getType() {
		return type;
	}

	public void setType(String type) {
		this.type = type;
	}

	public String getLocation() {
		return location;
	}

	public void setLocation(String location) {
		this.location = location;
	}

	public Boolean getOnSystem() {
		return onSystem;
	}

	public void setOnSystem(Boolean onSystem) {
		this.onSystem = onSystem;
	}

	public User getOwner() {
		return owner;
	}

	public void setOwner(User owner) {
		this.owner = owner;
	}

	public Set<User> getCollaborators() {
		return collaborators;
	}

	public void setCollaborators(Set<User> collaborators) {
		this.collaborators = collaborators;
	}

	public Set<History> getOperations() {
		return operations;
	}

	public void setOperations(Set<History> operations) {
		this.operations = operations;
	}

}

File: front-end/src/app/models/store.ts
import { Operation } from './Operation';
import { User } from "./user";
import { StoreProduct } from "./StoreProduct";

export class Store {
    private id: number;
	private name: string;
	private type: string;
	private location: string;
	private onSystem: boolean;
	
	private storeProducts: StoreProduct;
    private owner: User;
	
	private collaborators: User[];
	
	private operations: Operation[];
  
  

    /**
     * Getter $type
     * @return {string}
     */
	public get $type(): string {
		return this.type;
	}

    /**
     * Setter $type
     * @param {string} value
     */
	public set $type(value: string) {
		this.type = value;
	}

    /**
     * Getter $location
     * @return {string}
     */
	public get $location(): string {
		return this.location;
	}

    /**
     * Setter $location
     * @param {string} value
     */
	public set $location(value: string) {
		this.location = value;
	}

    /**
     * Getter $onSystem
     * @return {boolean}
     */
	public get $onSystem(): boolean {
		return this.onSystem;
	}

    /**
     * Setter $onSystem
     * @param {boolean} value
     */
	public set $onSystem(value: boolean) {
		this.onSystem = value;
	}

    /**
     * Getter $storeProducts
     * @return {StoreProduct}
     */
	public get $storeProducts(): StoreProduct {
		return this.storeProducts;
	}

    /**
     * Setter $storeProducts
     * @param {StoreProduct} value
     */
	public set $storeProducts(value: StoreProduct) {
		this.storeProducts = value;
	}

    /**
     * Getter $owner
     * @return {User}
     */
	public get $owner(): User {
		return this.owner;
	}

    /**
     * Setter $owner
     * @param {User} value
     */
	public set $owner(value: User) {
		this.owner = value;
	}

    /**
     * Getter $collaborators
     * @return {User[]}
     */
	public get $collaborators(): User[] {
		return this.collaborators;
	}

    /**
     * Setter $collaborators
     * @param {User[]} value
     */
	public set $collaborators(value: User[]) {
		this.collaborators = value;
	}

    /**
     * Getter $operations
     * @return {Operation[]}
     */
	public get $operations(): Operation[] {
		return this.operations;
	}

    /**
     * Setter $operations
     * @param {Operation[]} value
     */
	public set $operations(value: Operation[]) {
		this.operations = value;
	}

    /**
     * Getter $name
     * @return {string}
     */
	public get $name(): string {
		return this.name;
	}

    /**
     * Setter $name
     * @param {string} value
     */
	public set $name(value: string) {
		this.name = value;
	}

    /**
     * Getter $id
     * @return {number}
     */
	public get $id(): number {
		return this.id;
	}

    /**
     * Setter $id
     * @param {number} value
     */
	public set $id(value: number) {
		this.id = value;
	}
    constructor () {

    }
    setData(Data): void{
      this.id = Data.id;
      this.name = Data.userName;
      this.type = Data.type;
      this.location = Data.location;
      this.onSystem = Data.onSystem;
      this.storeProducts = Data.StoreProducts;
      this.owner = Data.owner;
      this.collaborators = Data.collaborators;
      this.operations = Data.operations;
    }
  }
  